### PR TITLE
dev db:seed - use addresses from CSV file (based on ENV variable)

### PR DESCRIPTION
**Pull request by weedySeaDragon**
### PT Story:  speed up db:seed - get already coded addresses from CSV file
https://www.pivotaltracker.com/story/show/162595079

### Changes proposed in this pull request:
1. Use already created and geocoded addresses to seed the _development_ db by default.
2. Define and use ENV variable `SHF_SEED_FAKE_ADDR_CSV_FILE` to specify a CSV file to use for addresses
3. If there are not enough addresses in the CSV file,  create (and geocode) brand new addresses
1.  Read (and write) Addresses from a CSV file ( `lib/fake_addresses/csv_fake_addresses_reader.rb` and `lib/fake_addresses/csv_fake_addresses_writer.rb`
2. create a CSV file of 89 addresses that are already geocoded (`db/fake-addresses-89--2018-12-12.csv`)  (89 because that's what I happened to have in my db at the time)
3. Refactored `seed_helpers.rb` so that addresses are created from the `AddressFactory` class.  (because seed_helpers.rb is a bit out of control. not OO; hard to test)
4. Added comments to `.env.development.example` and re-organized it a bit

#### Simple timing testing done on my machine:
Doing entire db_recreate with this script:
```
#!/usr/bin/env bash
echo "=========================================="
echo " Starting shf:db_recreate...."
SECONDS=0.0
RAILS_ENV=development bundle exec rake shf:db_recreate
duration=$SECONDS
echo ""
echo "Duration = $(($duration / 60)) minutes and $(($duration % 60)) seconds"
```

using addresses from CSV: (25 users, 22 applications): 8 seconds
creating all new addresses: (25 users, 22 applications): 1 min 24 seconds

Ready for review:
@patmbolger

### CODE REVIEW

File: db/seed_helpers.rb [hunk 1]
```diff
@@ -179,14 +187,154 @@ def make_app(user)
...
+    # remove it from the list of already constructed addresses
+    # and save it.
+    #
+    # If we cannot get an address, return nil
+    #
+    # We ensure that each address is used just once by
+    # removing it from the list of already constructed addresses.
+    #
+    # @param addressable_entity [] - the addressable object that we will associate with the address
+    # @return [Address] - an address that is _not_saved
```
Comment (patmbolger) [inline db/seed_helpers.rb]: ``` ... _not_ validated.```

**Review by patmbolger — APPROVED** [review]
Great work!  One minor comment.

**Comment by weedySeaDragon** [comment]
**Whoa!**   I just stepped back into the whole "mysterious list of lots of historical commits" thing!

I'm gonna revert the branch and see what happened.  (I think it had to do with some rebasing I did.)

Looks like my `origin` branch of `develop` (https://github.com/weedySeaDragon/shf-project)  has that list of 'historical' commits.  If I merge my `origin` branch into my current work (e.g. the branch for this PR), then I add that list of commits back in.
So: when I start and work with a branch for a PR, I should use `upstream` ( = AgileVentures/shf-project `develop` branch) and _not_ merge my `origin` into it.    And at some point I'll need to clean up my `origin` branch.

**Comment by weedySeaDragon** [comment]
@patmbolger - updated the comment to say:
```
# @return [Address] - an address that is saved but _not_ validated
```

(not sure you'd see the change easily due to my blorfed up commits)

Merge at your leisure `||` send along other good catches

**Pull request merged**